src/components/PaginatorPages.vue

```diff
@@ -0,0 +1,51 @@
+<template>
+  <div class="page">
+    <div class="fixed bottom-0 w-full">
+      <paginator-on 
+        :rows="20" 
+        :totalRecords="totalRecords" 
+        :first="(currentPage - 1) * 20"
+        :pageCount="maxPages"
+        template="FirstPageLink PrevPageLink CurrentPageReport NextPageLink LastPageLink"
+        currentPageReportTemplate="{first} até {last} de {totalRecords}" 
+        @page="onPageChange"
+        :pt="{ 
+          root: {
+            class:['bg-gray-800','text-sm md:text-base','rounded-none', 'text-slate-300', 'py-1 lg:py-0']
+          },
+          current: ['text-slate-300'],
+          firstpagebutton: ['hover:bg-gray-700'], previouspagebutton: ['hover:bg-gray-700'], nextpagebutton: ['hover:bg-gray-700'], lastpagebutton: ['hover:bg-gray-700']
+         }"
+        /> 
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+/* eslint-disable */
+import { defineComponent } from 'vue';
+
+export default defineComponent({
+  name: 'PaginatorPages',
+  props: {
+    totalRecords: {
+      type: Number,
+      required: true
+    },
+    currentPage: {
+      type: Number,
+      required: true
+    },
+    maxPages: {
+      type: Number,
+      required: true
+    }
+  },
+  methods: {
+    onPageChange(event: any): void {
```

Comment:
Put the coorect type for this event. On Api on primevue component web site there are the right types for each emit. For this case  
![Captura de tela de 2024-07-15 14-00-16](https://github.com/user-attachments/assets/38195c9d-5c23-4618-b03f-5464c80c57df)